File: hphp/hack/src/hhbc/hhas_typedef.ml
(*
 * Copyright (c) 2017, Facebook, Inc.
 * All rights reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the "hack" directory of this source tree.
 *
 *)

type t = {
  typedef_name: Hhbc_id.Class.t;
  typedef_attributes: Hhas_attribute.t list;
  typedef_type_info: Hhas_type_info.t;
  typedef_type_structure: Typed_value.t option;
}

let make
    typedef_name typedef_attributes typedef_type_info typedef_type_structure =
  {
    typedef_name;
    typedef_attributes;
    typedef_type_info;
    typedef_type_structure;
  }

let name hhas_typedef = hhas_typedef.typedef_name

let attributes hhas_typedef = hhas_typedef.typedef_attributes

let type_info hhas_typedef = hhas_typedef.typedef_type_info

let type_structure hhas_typedef = hhas_typedef.typedef_type_structure

File: hphp/hack/src/hhbc/hhas_typedef.rs
// Copyright (c) Facebook, Inc. and its affiliates.
//
// This source code is licensed under the MIT license found in the
// LICENSE file in the "hack" directory of this source tree.

use hhas_attribute_rust as hhas_attribute;
use hhas_type;
use hhbc_id_rust as hhas_id;
use runtime;

use hhas_attribute::HhasAttribute;
use runtime::TypedValue;

#[derive(Debug)]
pub struct Typedef<'a> {
    pub name: hhas_id::class::Type<'a>,
    pub attributes: Vec<HhasAttribute>,
    pub type_info: hhas_type::Info,
    pub type_structure: TypedValue,
}
